packages/teleterm/src/ui/services/usage/usageService.ts

```diff
@@ -0,0 +1,192 @@
+      },
+    });
+  }
+
+  captureAccessRequestCreate(
+    uri: ClusterOrResourceUri,
+    kind: 'role' | 'resource'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      accessRequestCreate: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+        kind,
+      },
+    });
+  }
+
+  captureAccessRequestReview(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      accessRequestReview: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+      },
+    });
+  }
+
+  captureAccessRequestAssumeRole(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      accessRequestAssumeRole: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+      },
+    });
+  }
+
+  captureFileTransferRun(
+    uri: ClusterOrResourceUri,
+    direction: 'download' | 'upload'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      fileTransferRun: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+        direction,
+      },
+    });
+  }
+
+  captureUserJobRoleUpdate(jobRole: string): Promise<void> {
+    return this.reportNonAnonymizedEvent({
+      userJobRoleUpdate: {
+        jobRole,
+      },
+    });
+  }
+
+  private reportNonAnonymizedEvent(
+    prehogEventReq: PrehogEventReq
+  ): Promise<void> {
+    return this.reportEvent('', prehogEventReq);
+  }
+
+  private reportEvent(
+    authClusterId: string,
+    prehogEventReq: PrehogEventReq
+  ): Promise<void> {
+    const isCollectingUsageMetricsEnabled = this.configService.get(
+      'usageMetrics.enabled'
+    ).value;
+
+    if (!isCollectingUsageMetricsEnabled) {
+      return;
+    }
+
+    return this.tshClient.reportUsageEvent({
+      authClusterId,
+      prehogReq: {
+        distinctId: this.runtimeSettings.installationId,
+        timestamp: new Date(),
+        ...prehogEventReq,
+      },
+    });
+  }
+
+  private getClusterProperties(uri: ClusterOrResourceUri) {
+    const rootClusterUri = routing.ensureRootClusterUri(uri);
+    const cluster = this.findCluster(rootClusterUri);
+    if (!(cluster && cluster.loggedInUser)) {
+      // TODO: add check for authClusterId
+      this.logger.warn(`Missing cluster data for ${uri}, skipping event`);
```

Comment:
I'd move this log higher up, at the place which actually skips reporting the event. The return in this method merely returns null cluster properties.

Reply:
Done